app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Admin::VideosController < AdminsController
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(video_params)
+
+    if @video.save
+      flash[:success] = "You have successfully added the video #{@video.title}"
+      redirect_to new_admin_video_path
+    else
+      flash[:error] = "Error - The video was not created."
```

Comment:
When you render, then you should use `flash.now` otherwise it carries to next request.